File: frontend-marketmotor/src/app/models/responseapi/ProductoResponse.ts
import { Producto } from "../dtos/Producto";

export class productoResponse{
    content: Producto[];
    pageNo: number;
    pageSize: number;
    totalElements: number;
    totalPages: number;
    last: boolean;

    constructor(){
        this.content  = []
        this.pageNo = 0;
        this.pageSize = 0;
        this.totalElements = 0;
        this.totalPages = 0;
        this.last = false;
    }
}

File: marketmotormicroservices/producto-service/src/main/java/com/jma/productoservice/producto/domain/response/ProductoResponse.java
package com.jma.productoservice.producto.domain.response;


import com.jma.productoservice.producto.domain.dto.ProductoDto;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ProductoResponse {

    private List<ProductoDto> content;
    private int pageNo;
    private int pageSize;
    private long totalElements;
    private int totalPages;
    private boolean last;
}
